v2.0 release

siduck opened the pull request:
https://github.com/orgs/NvChad/projects/8

Other pr's related to this release

https://github.com/NvChad/ui/pull/51
https://github.com/NvChad/base46/pull/156

Rewrite whole site in solidjs + vite, remove docusaurus cuz its a lil slow and bloated

https://github.com/NvChad/nvchad.github.io/pull/118

Edit : v2.0 has released and this PR will merge after 3-4 months when users have migrated to v2.0 properly

siduck committed:
add glyph icon for untracked icon (gitsigns

siduck committed:
use default mapping for diagnostic goto_next

siduck committed:
rm support for ooold plugin syntax

siduck committed:
remove alpha plugin

siduck committed:
use dev branch for ui/base46 plugins

this is temporary, gets easy to test dev changes locally

siduck committed:
move ui plugin config to chadrc UI table

siduck committed:
rm un-needed lazyload function

siduck committed:
change highlight loading function name

siduck committed:
add cmd to compile nvchad theme

siduck committed:
fix typo in mkdir cmd

siduck committed:
rm incorrect theme cache path

siduck committed:
fix packer bootstrapping not working

siduck committed:
rm title option

fixes (#1653) and this is a cmp issue + only with some terminals

siduck committed:
load base46_cache in instead of minified modules

siduck committed:
handle error base46_cache error

siduck committed:
rm un-needed cmd

siduck committed:
load syntax highlights in treesitter config

siduck committed:
add multiple cmp themes https://github.com/NvChad/base46/issues/109

siduck committed:
load bg file (containing vim.opt.bg value)

siduck committed:
rm outdated nvchad_ui setup call

siduck committed:
add statusline theme option

siduck committed:
add new dir icons for nvimtree

siduck committed:
add helper comments for statusline options

siduck committed:
set base46 as opt plugin

since we are now just using this plugin for compiling nvchad themes, we dont need this at startup

siduck committed:
use lua api to set commands

siduck committed:
use lua api for packersync wrapper

Commit:
ensure errors arent silently swallowed in custom chadrc if exists

siduck committed:
use dev branch of extensions repo

sha5010 committed:
config(cmp): simplify disabling scrollbar

siduck committed:
clean load_config function

based on @ii14 's comment https://github.com/NvChad/NvChad/pull/1704#discussion_r1040655958

siduck committed:
avoid swallowing custom/init.lua errors by pcall

siduck committed:
add option for cheatsheet theme

and add description for some mappings

siduck committed:
rm mappings for navigating within insert mode

describes the issue with it https://github.com/NvChad/NvChad/issues/1294#issuecomment-1359897212

lucario387 committed:
Fix mkdir on windows

siduck committed:
add function to install chadrc tempalte during bootstrap

siduck committed:
bootstrap chadrc template before packer boostrap

lucario387 committed:
Move friendly-snippets to start
There is no reason to lazy load this plugin as there is no lua file
Move it to start to simply loading sequence

lucario387 committed:
update nvim-cmp loading strategy

siduck committed:
migrate to lazy.nvim

siduck committed:
rm un-needed lazy.nvim config

those are defaults so no need to add them

siduck committed:
update update_branch variable to lazy

siduck committed:
include downloading of lazy-lock.json while bootstrapping

also pin all plugins by default

siduck committed:
avoid using vim.cmd! for running os commands

siduck committed:
rm unneeded whitespace in lazy-lock url

siduck committed:
rm un-needed cmdefinitons

anosillus committed:
remove kyazdani42 repo

siduck committed:
clean config & rm un-needed pcalls

siduck committed:
use build opt instead of run

ynhhoJ commented:
When is the CHAD release coming out? 👀👀👀

siduck commented:
> When is the CHAD release coming out? eyeseyeseyes

within next 2 weeks most likely, only thing to now do is finish writing the site and probably i'll do some presentations on youtube :nerd_face:

siduck committed:
update example_config branch in bootstrap module

siduck committed:
dont load statusline highlights cache

as its included in the defaults file now!

siduck committed:
use dofile to load cached theme files

siduck committed:
reduce some LOC

siduck committed:
add nvchad_types to sumenko lsp config

also remove nvimtree depreceated option

Commit:
update nvim-tree option names

ariefra committed:
Fix sumneko_lua deprecation, change to lua_ls

siduck committed:
add autocmd for automatically compiling base46

siduck committed:
add windows support for compile theme autocmd

siduck committed:
BREAKING CHANGE: utilize lazy.nvim fully, remove un-needed functions

siduck committed:
add condition to merge user custom plugins

siduck committed:
avoid loading cached highlights in plugin opts file

siduck committed:
wrap some opt modules with function

not doing so will make those modules load by default which isnt good

siduck committed:
misc

lucario387 committed:
Add lazy.nvim to workspace library

cliffordfajardo commented:
Just saw this Repo on my newsfeed; been wanting to get into Neovim for a while; glad to try this out as a first time neovim user!

siduck committed:
misc

siduck committed:
load cached highlights whenever base46 updates

siduck committed:
load mappings after custom init.lua

this ensures that custom leader key will work properly

siduck committed:
misc & cleanup

siduck committed:
exclude indentline in nvdash/nvcheatsheet fts

siduck committed:
put all core stuff in core.init only

siduck committed:
re-add reloading of all custom files

siduck committed:
run load_config() before reloading highlights onsaving chadrc

siduck committed:
add option for telescope style

siduck committed:
rm useless badge

siduck committed:
add mappings btn in nvdash

siduck committed:
add mappings btn in nvdash

siduck committed:
Update image links

siduck committed:
re-add ESC mapping for clearing highlights

siduck committed:
use main branch example_config in bootstrap

siduck committed:
Fix broken site links in readme (#1814)

siduck committed:
Update README.md

BrunoKrugel committed:
Update default_config.lua

siduck committed:
keep whichkey enabled by default

siduck committed:
temporarily silence dofile syntax error during bootstrap

we need to find a way to fetch base46 first and then rest of the plugins (during bootstrap)

siduck committed:
add options to disable signature

siduck committed:
make statusline separators reloadable

siduck committed:
add an option to disable tabufline

lucario387 committed:
fix #1832

siduck committed:
add variable for path seperators

lucario387 committed:
normalize all paths

lucario387 committed:
remove unused var

lucario387 committed:
fix NVIM_APPNAME

Diogo-ss committed:
fix: text in readme.md

mikededo commented on .github/README.md:
```diff
@@ -1,26 +1,20 @@
-            src="https://img.shields.io/github/license/NvChad/NvChad?style=flat-square&logo=GNU&label=License&color=df967f"
-            alt="License"
-    />
 [![Neovim Minimum Version](https://img.shields.io/badge/Neovim-0.8.0-blueviolet.svg?style=flat-square&logo=Neovim&color=90E59A&logoColor=white)](https://github.com/neovim/neovim)
```
Docs say minimum version is `0.8.3`. Which one should stay?

siduck replied:
@mikededo need to update the readme, thx for correcting!

mikededo replied:
I was looking through the docs today and I found a couple of improvements, do you want me to later create a PR with it? I would be willing to help with the docs 💪

siduck replied:
@mikededo feel free to!  Btw do know that all the docs are in md files, `/docs and /news` format.

siduck committed:
update nvim version in readme badge

siduck committed:
rm outdated info in readme

siduck committed:
luasnip: add globals for various types of snippets

siduck committed:
save nvchad cache at nvim data dir

many a time the user clears the cache dir and that results in deletion of base46 cache too which would prevent nvchad to function

siduck committed:
boostrap mason packages properly

print some message when nvchad is totally setup

siduck committed:
reduce loc in boostrap.lua

siduck committed:
give priority to defaults hl groups cache

fixes https://github.com/NvChad/base46/issues/173, not adding this will let plugins like vim-matchup messup our syntax highlight groups

siduck committed:
just load default bas46 cache instead of giving it lazy priority

https://github.com/NvChad/base46/issues/173

siduck committed:
load defaults cache after lazypath check

not having this will create that dofile error that the file isnt present as the file gets created only when base46 compiles and which occurs during boostrapping lazy.nvim

siduck committed:
bootstrap base46 before lazy

fix https://github.com/NvChad/base46/issues/166

siduck committed:
bootstrap: clone base46 to lazy path instead of tmpdir

siduck committed:
bootstrap.lua: rm useless variables

Daniel-Boll committed:
fix(cmp): lsp suggestions before snippets

siduck committed:
add nvimtree_side global in nvimtree config func https://github.com/NvChad/ui/issues/68

also clean some lazy.nvim related variables

siduck committed:
add show_numbers option to tabufline config

https://github.com/NvChad/ui/commit/87525fa3cd330b594261de7b63976ee66909af43

Daniel-Boll committed:
feat(bootstrap): indicate that no is the default answer

Indicates that the default answer would be *no* if the user provide any input other than `y`

waneon committed:
feat: Increase visibility of floating diagnostic

siduck committed:
use function in init opts for all plugins

if this isnt done then disabling a plugin will still have its init opt called, read https://github.com/NvChad/NvChad/discussions/1879

asher-gh committed:
gitsign: handling file path with spaces

siduck committed:
add nvimtree option to preserve window proportions (#1597)

Commit:
fix(mappings): fix description of j key in x mode

prometheusalpha committed:
feat(mappings): add description to movement keys in normal mode

siduck committed:
make some mapping keybinds lowercased (#1897)

sk02241994 committed:
Added current buffer fuzzy find key mapping (#1903)

* Added current buffer fuzzy find key mapping

* change telescope fuzzy mapping to <leader>fz

 Co-authored-by: Sidhanth Rathod <siduck@tutanota.com>

siduck committed:
add an option for lsp_semantic_tokens hl groups | fix (#1907)

also tiden up some stuff

siduck committed:
handle semantic_tokens on lsp_attach based on chadrc option (#1907)

siduck committed:
clean up

siduck committed:
rm depcreated option from nvimtree config

siduck committed:
include c , v keys in whichkey lazyloading

siduck commented:
I just realized we dont need this PR, all versions of nvchad will stay in their own branches!
